src/evidently/collector/app.py

```diff
@@ -46,14 +46,12 @@
 logger = logging.getLogger(__name__)
 
 
-@post(
-    "/{id:str}",
-)
-async def create_collector(
+@post("/{id:str}", sync_to_thread=True)
+def create_collector(
```

Comment:
I've changed these to synchronous endpoints with `sync_to_thread=True` because they were all performing some sort of blocking operation, usually related to the `CollectorService`/`CollectorStorage`; They both perform blocking IO under the hood, which means that, when called from within an async function like this, they will block the event loop for the whole duration, effectively preventing any other requests from running concurrently. (some further reading can be done [here](https://docs.litestar.dev/latest/topics/sync-vs-async.html))

The easy fix here is to use synchronous endpoints and let Litestar handle the concurrency, but a more elegant way to solve this would be to make the underlying APIs asynchronous as well. From looking at the code briefly, this shouldn't be too much of an issue, but I didn't know if they were external dependencies on these / APIs that would break were those changes introduces.